src/commands/case_management.py

```diff
@@ -0,0 +1,732 @@
+    return " ".join(words)
+
+
+@command("list")
+async def cmd_list(ctx: Context):
+    """
+    Implementation of !list
+
+        Supported parameters:
+        -i: Also show inactive (but still open) cases.
+        -r: Show assigned rats
+        -u: Show only cases with no assigned rats
+        -@: Show full case IDs.  (LONG)
+
+    Args:
+        ctx:
+
+    """
+    _, *words = ctx.words
+
+    flags = ListFlags()
+    platform_filter = None
+
+    # plain invocation
+    if len(words) == 0:
+        ...  # use above defaults (done this way so else can be used below as an error state)
+
+    # arguments invocation
+    elif len(words) == 1 or len(words) == 2:
+        flags_set = False
+        platform_filter_set = False
+
+        for word in words:  # type: str
+            if word.startswith("-"):
+                if flags_set:
+                    raise RuntimeError("invalid usage")  # FIXME: usage warning to user
+                flags = ListFlags.from_word(word)
+                flags_set = True
+            else:
+                # platform or bust
+                if platform_filter_set:
+                    raise RuntimeError("invalid usage")  # FIXME: usage error
+
+                try:
+                    platform_filter = Platforms[word.upper()]
+                except KeyError:
+                    return await ctx.reply(f"unrecognized platform '{word.upper()}'")
+
+    else:
+        raise RuntimeError  # FIXME: usage error
+    logger.debug(f"flags set:= {flags} \t platform_filter := {platform_filter}")
+    active_rescues: typing.List[Rescue] = []
+    inactive_rescues: typing.List[Rescue] = []
+
+    rescue_filter = functools.partial(_rescue_filter, flags, platform_filter)
+
+    # for each rescue that doesn't matches the filter
+    for rescue in itertools.filterfalse(rescue_filter, iter(ctx.bot.board.values())):  # type: Rescue
+        # put it in the right list
+        if rescue.active:
+            active_rescues.append(rescue)
+        else:
+            inactive_rescues.append(rescue)
+    format_specifiers = "c"
+    if flags.show_assigned_rats:
+        format_specifiers += "r"
+    if flags.show_uuids:
+        format_specifiers += "@"
+
+    if not active_rescues:
+        await ctx.reply("No active rescues.")
+    else:
+
+        output = _list_rescue(active_rescues, format_specifiers)
+        if output:
+            await ctx.reply(output)
+    if flags.show_inactive:
+        if not inactive_rescues:
+            return await ctx.reply("No inactive rescues.")
+
+        output = _list_rescue(inactive_rescues, format_specifiers)
+        if output:
+            await ctx.reply(output)
+
+
+def _list_rescue(rescue_collection, format_specifiers):
+    buffer = io.StringIO()
+    buffer.write(f"{len(rescue_collection)} active cases. ")
+    for rescue in rescue_collection:
+        buffer.write(format(rescue, format_specifiers))
+        buffer.write("\n")
```

Comment:
[Error?] Looking at Mecha2's current `list` functionality, isn't there only meant to be a newline between cases if additional flags are given? (e.g. show assigned rats)